tools/provenance-generator/main.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// release-jobs-syncer fetches latest release branches,
+// and creates PRs to update Prow jobs for them in knative/test-infra
+
+package main
+
+import (
+	"flag"
+	"log"
+
+	"knative.dev/test-infra/tools/provenance-generator/pkg"
+)
+
+func main() {
+	cloneLogPath := flag.String("clone-json", "", "Path to clone.json")
+	if cloneLogPath == nil {
+		log.Fatal("--clone-json cannot be empty")
+	}
+
+	imageRefPath := flag.String("image-refs", "", "Path to image references generated by ko")
+	if imageRefPath == nil {
+		log.Fatal("--image-refs cannot be empty")
+	}
+	flag.Parse()
+
+	config := pkg.Config{
+		CloneLogPath:           *cloneLogPath,
+		EntryPointOptsVariable: "ENTRYPOINT_OPTIONS",
+		ImageReferencePath:     *imageRefPath,
+	}
+
+	config = pkg.LoadParameters(config)
+	pkg.GenerateAttestation(config)
```

Comment:
Minor style point -- I'd have `GenerateAttestation` return a `provenance.Statement`, and then do the output writing here. It will make it easier to test in the future, I expect.